tiantianXia/assignments/mergeIntervals/lc57/lc57.js

```diff
@@ -16,33 +52,66 @@ class Interval {
 const insert = function (intervals, new_interval) {
   let merged = [];
   // TODO: Write your code here
+  let start = new_interval.start;
+  let end = new_interval.end;
+  let i = 0;
+  let isOverlapping = false;
+
+  while(i < intervals.length || isOverlapping){
+    if(i === intervals.length) i -= 1;
+    if(start <= intervals[i].start && end >= intervals[i].start){
+        end = Math.max(end, intervals[i].end);
+        isOverlapping = true;
+    }else{
+      if(start <= intervals[i].end){
+        start = intervals[i].start;
+        end = Math.max(end, intervals[i].end);
+        isOverlapping = true;
+      }
+    }
+    if(!isOverlapping){
+      merged.push(new Interval(intervals[i].start, intervals[i].end));
+    }else if (i < intervals.length - 1){
+      if(end <= intervals[i + 1].start){
+        merged.push(new Interval(start, end));
+        isOverlapping = false;
+      }
+    }else if(i === intervals.length - 1){
```

Comment:
} else if  ( ...  ) ---> spacing will become increasingly important for more and more complex solutions